### ISSUE TITLE
0.9.0-rc5 make program stuck on slurm+ddp, when using Accuracy

### ISSUE THREAD

**Issue by xiadingZ** [PyTorchLightning/pytorch-lightning#2901]
## 🐛 Bug

after upgrading to 0.9.0-rc5 from rc4, my program stucks on val loop, no error message. train loop is normal.
all difference between train/val loop is my val loop use 'Accuracy' TensorMetric

I'm not sure it's because of 'Accuracy' or bugs in 'validation_step'.
my program use ddp + slurm, normal in 0.9.0-rc4

 - PyTorch Version (e.g., 1.0): 1.5.0
 - Any other relevant information: slurm

**Comment by Borda** [comment]
@awaelchli is working on it... :]

**Comment by awaelchli** [comment]
Yes, I suspect it's the same issue I'm fixing here #2790 . Have not tested it with slurm yet, but if you want you can already try my fix:
`pip install git+https://github.com/PyTorchLightning/pytorch-lightning.git@bugfix/test-before-fit`

**Comment by xiadingZ** [comment]
It's not working with slurm yet, on my environment

**Comment by williamFalcon** [comment]
mind checking with master?

**Comment by xiadingZ** [comment]
No, it still doesn't work on version >= 0.9.0-rc5, stuck on validation.   normal on 0.9.0-rc4.
![截屏2020-08-17 上午10 37 21](https://user-images.githubusercontent.com/16729275/90352476-17e70400-e076-11ea-927b-9204d4d2cbda.png)

**Comment by williamFalcon** [comment]
use rc15

**Comment by xiadingZ** [comment]
I just tried master, rc15. doesn't work

**Comment by xiadingZ** [comment]
sorry, I find It's my problem. If I set a lower batch-size, it works on rc15
